src/serve/index.ts

```diff
@@ -0,0 +1,48 @@
+import { EmulatorServer } from "../emulator/emulatorServer";
+
+const _ = require("lodash");
+
+const logger = require("../logger");
+
+const TARGETS: {
+  [key: string]:
+    | EmulatorServer
+    | { start: (o: any) => void; stop: (o: any) => void; connect: () => void };
+} = {
+  hosting: require("./hosting"),
+  functions: require("./functions"),
+  database: require("./database"),
+  firestore: require("./firestore"),
+};
+
+const _serve = async function(options: any) {
```

Comment:
Prefer `export async function serve(options: any)` over this pattern. This will require changing the locations where `serve` is imported in other files (which is acceptable).